### PR TITLE
Changed API specifications to be up-to-date and pull request template

### PR DESCRIPTION
**Pull request by IsaacLambat**
- Changed API specifications to match current database config options
- Added summary to endpoint without one so it would not show the description as the title
- Updated redocly config to use `apis` instead of `apiDefinitions` which is deprecated
- Updated PR template

### CODE REVIEW

File: docs/api/components/schemas.yaml [hunk 1]
```diff
@@ -1002,11 +1025,17 @@ Database:
       description: The maximum amount of query operations that can be running at any one point.
       type: integer
       default: 1000
+    scopes:
+      description: Scope and collection specific config.
+      type: object
+      properties:
+        scope_name:
+          $ref: '#/Scopes'
```
Comment (bbrks) [inline docs/api/components/schemas.yaml]:
```suggestion
      additionalProperties:
        $ref: '#/Scopes'
```

File: docs/api/components/schemas.yaml [hunk 2]
```diff
@@ -963,6 +963,29 @@ Replication-status:
   required:
     - replication_id
   title: Replication-status
+Scopes:
+  description: A map of all the collections with their corrosponding configs for this scope
+  type: object
+  properties:
+    collection_name:
+      $ref: '#/CollectionConfig'
```
Comment (bbrks) [inline docs/api/components/schemas.yaml]: `collection_name` isn't a literal, should be a user-definable property that has a CollectionsConfig type

```suggestion
    collections:
      description: The set of collections on the scope
      type: object
      additionalProperties:
        $ref: '#/CollectionConfig'
```

File: docs/api/components/schemas.yaml [hunk 3]
```diff
@@ -963,6 +963,29 @@ Replication-status:
...
+  title: Scopes
+CollectionConfig:
+  description: The configuration for the individual collection
+  type: object
+  properties:
+    SyncFn:
+      description: The Javascript function that newly created documents in this collection are ran through.
+      type: string
+      default: 'function(doc){channel(doc.channels);}'
```
Comment (bbrks) [inline docs/api/components/schemas.yaml]: This won't actually be the default sync function for a collection in its final implementation. Just keep it as an example.

```suggestion
      example: 'function(doc){channel(doc.channels);}'
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -3,14 +3,14 @@ CBG-0000
 Describe your PR here...
 - Use bullet points if there's more than one thing changed
 
-## Pre-review checklist (remove once done)
```
Comment (bbrks) [inline .github/PULL_REQUEST_TEMPLATE.md:6]: I think we should leave it here but just remove the `(remove once done)` part, and make each item a checkbox.

It really should be the developers responsibility to make sure this is done before it's considered ready for review, but can't harm to leave the list to remind the reviewer to check.

File: docs/api/components/schemas.yaml [hunk 5]
```diff
@@ -963,6 +963,29 @@ Replication-status:
   required:
     - replication_id
   title: Replication-status
+Scopes:
+  description: A map of all the collections with their corrosponding configs for this scope
```
Comment (bbrks) [inline docs/api/components/schemas.yaml]:
```suggestion
  description: A map of all the collections with their corresponding configs for this scope
```

**Review by bbrks — CHANGES_REQUESTED** [review]
1. I don't agree about absolving the developer of doing the checklist. But it should be left up as a reminder for reviewers to check these.

2. The scopes/collections parts need changing to reflect the API instead of static `collection_name` properties.

Once the `additionalProperties` is used instead, the specs become clearer that it's a user-definable property name:
 
Screenshot 2022-07-13 at 18 31 41

Screenshot 2022-07-13 at 18 31 27

File: docs/api/components/schemas.yaml (hunk 2 above)
Comment (IsaacLambat) [reply docs/api/components/schemas.yaml]: Makes sense. Forgot about that property. Saw a couple other places which didn't use this when they should have so will change them as well.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -3,14 +3,14 @@ CBG-0000
...
-- Removed debug logging (`fmt.Print`, `log.Print`, ...)
-- Logging sensitive data? Make sure it's tagged (e.g. `base.UD(docID)`, `base.MD(dbName)`)
-- Updated relevant information in the API specifications (such as endpoint descriptions, schemas, ...) in `docs/api`
-
 ## Dependencies (if applicable)
 - [ ] Link upstream PRs
-- [ ] Bump manifest once merged
+- [ ] Update Go module dependencies when merged
```
Comment (markspolakovs) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: ~~Is this step necessary? Manifests need manual work but if I've understood it correctly `go.mod` should be dealt with automatically by the Go tooling.~~

Never mind, realised you meant updating the go.mod once upstream PRs are merged - ignore me :D

**Review by bbrks — APPROVED** [review]

**Pull request merged**